back/boxtribute_server/cli/service.py

```diff
@@ -0,0 +1,158 @@
+from auth0 import Auth0Error
+from auth0.authentication import GetToken
+from auth0.management import Auth0
+
+from ..exceptions import ServiceError
+from .utils import setup_logger
+
+LOGGER = setup_logger(__name__)
+
+
+class Auth0Service:
+    def __init__(self, interface):
+        self._interface = interface
+
+    def get_users_of_base(self, base_id):
+        """Fetch all users of the given base. Return lists of users sorted by
+        single-base/multi-base type.
+        """
+        base_id = str(base_id)
+        # https://github.com/auth0/auth0-python/blob/6b1199fc74a8d2fc6655ffeef09ae961dc0b8c37/auth0/management/users.py#L55
+        users = []
+        try:
+            # Pagination setup
+            page = 0
+            per_page = 50
+            rest = None
+            while True:
+                response = self._interface.users.list(
+                    q=f'app_metadata.base_ids:"{base_id}" AND blocked:false',
+                    fields=["app_metadata", "user_id", "name"],
+                    page=page,
+                    per_page=per_page,
+                )
+                LOGGER.info(
+                    f"Fetched page {page + 1} of user data of total {response['total']}"
+                    " users."
+                )
+                if rest is None:
+                    rest = response["total"]
+                users.extend(response["users"])
+
+                # Pagination logic: go to next page; stop if nothing left
+                page += 1
+                rest -= per_page
+                if rest < 1:
+                    break
+        except Auth0Error as e:
+            raise ServiceError(code=e.status_code, message=e.message)
+
+        # Sort response into single- and multi-base users
+        result = {"single_base": [], "multi_base": []}
+        for user in users:
+            metadata = user["app_metadata"]
+            base_ids = metadata.get("base_ids", [])
+            if base_id not in base_ids:
+                LOGGER.warn(
+                    f"Base ID {base_id} not present in metadata base IDs: "
+                    f"{', '.join(base_ids)}"
+                )
+                continue
+            if len(base_ids) == 1:
+                result["single_base"].append(user)
+            else:
+                result["multi_base"].append(user)
+
+        return result
+
+    def get_single_base_user_role_ids(self, users):
+        errors = {}
```

Comment:
Please change so that the roles are identified via their name, e.g. all roles starting with base_1_

Reply:
This means we rely on the role name for identifying single-base roles, or to put it differently, we rely on the role name to derive a role attribute.

For your suggestion above we have to take care of updating the role of multi-base users, e.g. for a `base_1-2_coordinator`, after base 1 has been removed, they should become `base_2_coordinator`, and the `base_1-2_coordinator` role should not exist anymore. Or would they be `[base_1_coordinator, base_2_coordinator]` in the first place?